pandas/core/groupby/grouper.py

```diff
@@ -986,3 +989,29 @@ def _convert_grouper(axis: Index, grouper):
         return grouper
     else:
         return grouper
+
+
+def _is_same_memory(s1, s2):
+    if not is_dtype_equal(s1, s2):
+        return False
+    if len(s1) != len(s2):
+        return False
+
+    arr1 = s1._values
+    arr2 = s2._values
+
+    if isinstance(arr1, np.ndarray):
+        # slice first element -> if first element shares memory and they are
+        # equal length -> share exactly memory for the full length (if not
+        # checking first element, two arrays might overlap partially)
```

Comment:
are they necessarily contiguous?

Reply:
That's a good question (and we should cover that in tests). No, not necessarily I suppose. But the question is if that matters? 

In theory they could have a first same memory point, but if not contiguous and if they have different strides, they might still have different values. 
But can you get that in practice with a Series from a DataFrame?

Reply:
Can of course easily check that the strides are equal. But is that enough guarantee?